scene/2d/animated_sprite_2d.cpp

```diff
@@ -217,14 +218,22 @@ void AnimatedSprite2D::_notification(int p_what) {
 						}
 					}
 
+					timeout = _get_frame_duration();
+
 					queue_redraw();
 
 					emit_signal(SceneStringNames::get_singleton()->frame_changed);
 				}
 
-				double to_process = MIN(timeout, remaining);
+				float abs_speed_scale = Math::abs(speed_scale);
+				double to_process = MIN(timeout / abs_speed_scale, remaining);
+				timeout -= to_process * abs_speed_scale;
 				remaining -= to_process;
-				timeout -= to_process;
+
+				i++;
+				if (i > fc) {
+					return;
+				}
```

Comment:
This check seems unnecessary. Can you explain what it is doing?

Reply:
A limit on the number of iterations is needed to prevent freezes if, for some reason, `to_process` is each time much less than `remaining`. A similar protection is implemented here:

https://github.com/godotengine/godot/blob/44c55ad59b01dbf360777b1a9a18a229aeb1f4f7/scene/resources/texture.cpp#L2622-L2623

This is possible if all animation frames have zero or very short duration, or if the `speed_scale` is very large.

Reply:
Okay, makes sense. Then it would be better to add a comment or rename the variable `int fc` to `int iter_max`.

Reply:
Done.